microprofile/microprofile-fault-tolerance-2.1/src/main/java/org/wildfly/swarm/ts/microprofile/fault/tolerance/v21/retry/RetryService.java

```diff
@@ -0,0 +1,165 @@
+package org.wildfly.swarm.ts.microprofile.fault.tolerance.v21.retry;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.enterprise.context.RequestScoped;
+
+import org.eclipse.microprofile.faulttolerance.Asynchronous;
+import org.eclipse.microprofile.faulttolerance.Fallback;
+import org.eclipse.microprofile.faulttolerance.Retry;
+import org.wildfly.swarm.ts.microprofile.fault.tolerance.v21.MyConnection;
+import org.wildfly.swarm.ts.microprofile.fault.tolerance.v21.MyException;
+
+@Asynchronous // all is async with fallback method
```

Comment:
Technically `@Fallback` could also be defined here, because it's always the same.

Reply:
It could be nice but @Fallback is defined only for using on methods - @Target({ ElementType.METHOD })

Reply:
Ah, I didn't even know that, that tells something about me :-D Thanks!